rundeckapp/src/test/groovy/com/dtolabs/rundeck/server/plugins/logging/SimpleDataFilterPluginSpec.groovy

```diff
@@ -224,6 +224,52 @@ class SimpleDataFilterPluginSpec extends Specification {
 
     }
 
+    @Unroll
+    def "invalid characters replaced by custom character"() {
+        given:
+        def plugin = new SimpleDataFilterPlugin()
+        plugin.regex = regex
+        plugin.logData = false
+        plugin.name = name
+        plugin.invalidKeyPattern = validKeyPattern
+        plugin.replaceFilteredResult = true
+        plugin.invalidCharactersReplacement = invalidStringReplacement
+        def sharedoutput = new DataOutput(ContextView.global())
+        def context = Mock(PluginLoggingContext) {
+            getOutputContext() >> sharedoutput
+        }
+        def events = []
+        lines.each { line ->
+            events << Mock(LogEventControl) {
+                getMessage() >> line
+                getEventType() >> 'log'
+                getLoglevel() >> LogLevel.NORMAL
+            }
+        }
+        when:
+        plugin.init(context)
+        events.each {
+            plugin.handleEvent(context, it)
+        }
+        plugin.complete(context)
+        then:
+
+        sharedoutput.getSharedContext().getData(ContextView.global())?.getData() == (expect ? ['data': expect] : null)
+        if (expect) {
```

Comment:
unnecessary condition (?), expect is never null/false

Reply:
Done